src/nsrt_learning/side_predicate_learning.py

```diff
@@ -456,18 +495,68 @@ def _sideline(self) -> List[PartialNSRTAndDatastore]:
                     for a in pnad.op.preconditions
                 }
 
+        # Now that the add effects and preconditions are correct,
+        # make a list of all final PNADs. Note
+        # that these final PNADs only come from the
+        # param_opt_to_nec_pnad dict, since we can be assured
+        # that our backchaining process ensured that the
+        # PNADs in this dict cover all of the data!
```

Comment:
is there an easy way to assert this? (hmm, i can't think of one... because segments could be in multiple pnads' datastores... right?)

Reply:
we could run through all of datapoints in our training data and check that each one is associated with at least one PNAD?

Reply:
this would be a good check, but it doesn't really fit into the recompute_datastores API nicely...